exporter/collector/metrics.go

```diff
@@ -115,6 +138,12 @@ func (me *MetricsExporter) Shutdown(ctx context.Context) error {
 	go func() {
 		// Wait until all goroutines are done
 		me.goroutines.Wait()
+		// Close the WAL if open
+		if me.wal != nil {
+			if err := me.wal.Close(); err != nil {
+				me.obs.log.Error(fmt.Sprintf("error closing WAL: %+v\n", err))
+			}
```

Comment:
Consider moving this into into a defer in `runWALReadAndExportLoop()` so the goroutine can handle its own cleanup when it receives the shutdown signal.

Reply:
sgtm